File: backend/src/main/java/org/rent/room/be/service/PackageService.java
package org.rent.room.be.service;

import org.rent.room.be.entity.Package;

import java.util.List;
import java.util.UUID;

public interface PackageService {

    /**
     * Create a new Package
     * @param pkg package entity to create
     * @return created package
     */
    Package createPackage(Package pkg);

    /**
     * Get a package by id
     * @param id package uuid
     * @return found package
     */
    Package getPackageById(UUID id);

    /**
     * Get all packages
     * @return list of packages
     */
    List<Package> getAllPackages();

    /**
     * Update existing package
     * @param id package id
     * @param pkg package data to update
     * @return updated package
     */
    Package updatePackage(UUID id, Package pkg);

    /**
     * Delete package by id
     * @param id package id
     */
    void deletePackage(UUID id);
}

File: frontend/src/services/package/packageService.ts
import api from "../../config/axios";
import type { ApiResponse } from "../usersService";

// ===== INTERFACES =====
// Khớp 100% với RentPackageResponse.java ở BE

export interface PackageResponse {
  rentPackageId: string;       // UUID từ BE
  rentPackageName: string;
  price: number;
  durationDays: number;
  description: string;
  createdAt: string;           // LocalDateTime → string khi qua JSON
  updatedAt: string;
}

export interface PackageRequest {
  rentPackageName: string;
  price: number;
  durationDays: number;
  description: string;
}

// ===== SERVICE =====

export const packageService = {
  // Lấy tất cả gói — không cần đăng nhập (public)
  getAllPackages: () => {
    return api.get<ApiResponse<PackageResponse[]>>("/packages");
  },

  // Lấy chi tiết 1 gói theo id
  getPackageById: (id: string) => {
    return api.get<ApiResponse<PackageResponse>>(`/packages/${id}`);
  },

  // Tạo gói mới — chỉ ADMIN
  createPackage: (data: PackageRequest) => {
    return api.post<ApiResponse<PackageResponse>>("/packages", data);
  },

  // Cập nhật gói — chỉ ADMIN
  updatePackage: (id: string, data: PackageRequest) => {
    return api.put<ApiResponse<PackageResponse>>(`/packages/${id}`, data);
  },

  // Xóa gói — chỉ ADMIN
  deletePackage: (id: string) => {
    return api.delete<ApiResponse<void>>(`/packages/${id}`);
  },
};